tools/psoxy-test/README.md

```diff
@@ -1,13 +1,78 @@
+Node.js testing tool for Worklytics Psoxy.
 
-## Requirements
+This folder contains Node.js scripts to help you test your Worklytics Psoxy deploy. The requirements to be able to run the scripts are [Node.js] (version >=16) and [npm] (version >=8). First of all, install the npm dependencies: `npm i`.
 
-[Node.js](version >=16) and [npm] (version >=8).
+The primary tool is a command line interface (CLI) script that allows you to execute single "test calls" to your Worklytics Psoxy instance.  Check all the available options by running `node cli.js -h`. Those options may vary depending on whether you've deployed the Worklytics Psoxy to Amazon Web Services ([AWS]) or Google Cloud Platform ([GCP]).
 
-1. `npm i` 
-2. `node tools/psoxy-test/cli.js -h`
+## AWS
+Assuming that you've successfully deployed the Psoxy to AWS, and you've configured [Google Calendar] as data source, let's see an example:
+```shell
+node cli.js -u https://acme.lambda-url.us-east-1.on.aws/calendar/v3/calendars/primary -r arn:aws:iam::310635719553:role/PsoxyApiCaller -i user@acme.com
+```
+The `-r` option is mandatory for AWS deploys, and identifies the Amazon Resource Name (ARN) of the "role" that will be assumed (*) to be able to execute the call. The `-u` option is the URL you want to test. In this case, the URL's path matches a Google Calendar API endpoint (access the primary calendar of the currently logged-in user). The `-i` option identifies the user "to impersonate"; this option is only relevant for Google Workspace data sources.
 
+Another example for [Zoom]:
+```shell
+node cli.js -u https://acme.lambda-url.us-east-1.on.aws/v2/users -r arn:aws:iam::310635719553:role/PsoxyApiCaller
+```
+As you can see, the differences are:
+1. As this is not a Google Workspace data source, you don't need the `-i` option.
+2. The URL's path matches a [Zoom API endpoint] in this case
+
+(*) Requests to AWS API need to be [signed], so you must ensure that the machine running these scripts have the appropriate AWS credentials for the role you've selected.
+
+## GCP
+For GCP, every call must include the `-t` option, which is the identity token used for authentication (*). 
+Google Calendar example:
+```shell
+node cli.js -u https://us-central1-acme.cloudfunctions.net/psoxy-gcal/calendar/v3/calendars/primary -t <IDENTITY_TOKEN> -i user@acme.com
+```
+Zoom example:
+```shell
+node cli.js -u https://us-central1-acme.cloudfunctions.net/v2/users -t <IDENTITY_TOKEN>
+```
+
+(*) You can obtain it by running `gcloud auth print-identity-token` (using Google Cloud SDK)
+
+## Testing all endpoints for a given data source
+We also provide some npm scripts that allow you to test all the endpoints for a given data source. These scripts rely on the same Node.js module that runs the CLI script described above, but instead of providing the options via CLI, you have to create a `.env` file with the following contents:
+```shell
+# For AWS deploys
+PSOXY_URL=<url>
```

Comment:
i think this DEFINITELY needs to be an argument, because it will differ for each proxy invocation.

eg, gcal, gmail, gdrive psoxy instances are all distinct cloud functions in GCP, so will have a different value for  `PSOXY_URL` when you run the suite. Hence I think little value in defining in a flat file